### PR TITLE
Return key => val array of data from which the hash is built

### PR DESCRIPTION
**Pull request by S-anasol**
In my case i want store all fingerprint data in DB.
But I have not found anything that can do it in your lib :)

So here my fast realisation.

Maybe you can do this idea better or "right-way".

or you can just merge my code :)

### CODE REVIEW

**Comment by Valve** [comment]
This looks good to me, however you did some changes that I don't understand. Please see my comments.

File: fingerprint2.js [hunk 1]
```diff
@@ -780,7 +784,7 @@
         os = "Android";
       } else if(userAgent.indexOf("linux") >= 0){
         os = "Linux";
-      } else if(userAgent.indexOf("iphone") >= 0 || userAgent.indexOf("ipad") >= 0 ){
```
Comment (Valve) [inline fingerprint2.js]: Why making this change?

File: fingerprint2.js [hunk 2]
```diff
@@ -492,7 +496,7 @@
     },
     touchSupportKey: function (keys) {
       if(!this.options.excludeTouchSupport){
-        keys.push(this.getTouchSupport());
+        keys.push({key: "getTouchSupport", value: this.getTouchSupport()});
```
Comment (Valve) [inline fingerprint2.js]: these should have names without 'get' part, for example 'touch_support'.

File: fingerprint2.js [hunk 3]
```diff
@@ -234,36 +238,36 @@
         }
         return keys;
       }
-      keys.push(this.getWebglFp());
+      keys.push({key: "getWebglFp", value: this.getWebglFp()});
```
Comment (Valve) [inline fingerprint2.js]: try to use the short possible names, for example 'webgl' etc

File: fingerprint2.js (hunk 2 above)
Comment (Valve) [inline fingerprint2.js]: Please also make more descriptive commit messages. "change fingerprint.js" is a really bad commit message. It says nothing.

File: fingerprint2.js (hunk 1 above)
Comment (S-anasol) [inline fingerprint2.js]: i just used old version from composer, forgot to update it

File: fingerprint2.js (hunk 2 above)
Comment (S-anasol) [inline fingerprint2.js]: yeah i see.
I did Copy/Paste from right side :)

File: fingerprint2.js (hunk 3 above)
Comment (S-anasol) [inline fingerprint2.js]: Copy/Paste :smile:

**Comment by S-anasol** [comment]
Should be better now :pray:

**Comment by Valve** [comment]
I've merged, but please use 'contributing.md' next time. You obviously forgot to run gulp.

**Comment by Valve** [comment]
Please add a separate PR with a spec, covering your new functionality. After that I will release a new version.

**Pull request merged**